model-archiver/model_archiver/model_archiver_config.py

```diff
@@ -0,0 +1,40 @@
+import os
+from argparse import Namespace
+from dataclasses import dataclass
+from typing import Literal, Optional
+
+from model_archiver.manifest_components.manifest import RuntimeType
+
+
+@dataclass
+class ModelArchiverConfig:
+    model_name: str
+    handler: str
+    version: str
+    serialized_file: Optional[str] = None
+    model_file: Optional[str] = None
+    extra_files: Optional[str] = None
+    runtime: str = RuntimeType.PYTHON.value
+    export_path: str = os.getcwd()
+    archive_format: Literal["default", "tgz", "no-archive"] = "default"
+    force: bool = False
+    requirements_file: Optional[str] = None
+    config_file: Optional[str] = None
+
+    @staticmethod
+    def from_args(args: Namespace) -> "ModelArchiverConfig":
+        config = ModelArchiverConfig(
+            model_name=args.model_name,
```

Comment:
The parameters and fields  between args and config should always be aligned. Would be good to make the argument transfer automatic to remove one point that we need to touch in case we alter the parameters. Something like this:
```
from dataclasses import fields
# ...
     @classmethod
    def from_args(cls, args: Namespace) -> "ModelArchiverConfig":
        params = {field.name: getattr(args, field.name) for field in fields(cls)}
        config = cls(**params)
        return config
```